lib/Ed25519KeyPair.js

```diff
@@ -280,27 +281,12 @@ class Ed25519KeyPair extends LDKeyPair {
   /**
    * Generates and returns a multiformats encoded
    * ed25519 public key fingerprint (for use with cryptonyms, for example).
-   * @example
-   * > edKeyPair.fingerprint();
-   * z6dfdsfdsfds3432423
    * @see https://github.com/multiformats/multicodec
    *
-   * @returns {string} A verifiable cryptographic signature.
+   * @returns {string} The fingerprint.
    */
   fingerprint() {
-    // ed25519 cryptonyms are multicodec encoded values, specifically:
-    // (multicodec ed25519-pub 0xed01 + key bytes)
-    const pubkeyBytes = _base58Decode({
-      decode: base58.decode,
-      keyMaterial: this.publicKeyBase58,
-      type: 'public'
-    });
-    const buffer = new Uint8Array(2 + pubkeyBytes.length);
-    buffer[0] = 0xed;
-    buffer[1] = 0x01;
-    buffer.set(pubkeyBytes, 2);
-    // prefix with `z` to indicate multi-base base58btc encoding
-    return `z${base58.encode(buffer)}`;
+    return util.base58PublicKeyFingerprint({keyMaterial: this.publicKeyBase58});
```

Comment:
Wait, so, is there any chance we can try and not move functions to a generic `util` module? I get that on some rare occasions, there is absolutely no where to put a function. But in this case, there are all these files that make sense to put it in. Why move it to util? 
I'd much rather see this as either a class function on KeyPair, or just a plain function that's exported from this file. Or create a `fingerprint.js` module, or something.